File: android/app/src/main/java/com/didit/app/util/Apps.kt
package com.didit.app.util

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.DirectionsBike
import androidx.compose.material.icons.automirrored.filled.DirectionsRun
import androidx.compose.material.icons.filled.FitnessCenter
import androidx.compose.ui.graphics.vector.ImageVector

data class AppConfig(val label: String, val route: String, val icon: ImageVector)

/** Keyed by app kind ("running"/"biking"/"gym") — mirrors src/lib/apps.ts. */
val APPS: Map<String, AppConfig> = mapOf(
    "running" to AppConfig("Running", "running/home", Icons.AutoMirrored.Filled.DirectionsRun),
    "biking" to AppConfig("Biking", "biking/home", Icons.AutoMirrored.Filled.DirectionsBike),
    "gym" to AppConfig("Gym", "gym/home", Icons.Filled.FitnessCenter),
)

File: src/lib/apps.ts
import { Bike, Dumbbell, Footprints } from "lucide-react";

export type AppKind = "running" | "biking" | "gym";

export const APPS: Record<AppKind, { label: string; path: string; icon: typeof Bike }> = {
  running: { label: "Running", path: "/running", icon: Footprints },
  biking: { label: "Biking", path: "/biking", icon: Bike },
  gym: { label: "Gym", path: "/gym", icon: Dumbbell },
};
